### PR TITLE
Microsot.Network/virtualnetworkgateways Reset VPN Client Shared Key

### PR DESCRIPTION
**Pull request by henry416**
Added a reset VPN Client shared key operation (POST) on virtual network gateway resource.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by henry416** [comment]
@sergey-shandar Can you please review this PR?

**Review by sergey-shandar — APPROVED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json [hunk 1]
```diff
@@ -362,6 +362,49 @@
         "x-ms-long-running-operation": true
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworkGateways/{virtualNetworkGatewayName}/resetvpnclientsharedkey": {
```
Comment (sergey-shandar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json]: Do we have this operations in other API versions as well? 2018-07-01?

Comment (henry416) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json]: No, this is from 2018-06-01 onwards

**Comment by henry416** [comment]
@sergey-shandar Are there any other pending items? Can we merge?

**Review by sergey-shandar — APPROVED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json [hunk 2]
```diff
@@ -362,6 +362,49 @@
...
+      "post": {
+        "tags": [
+          "VirtualNetworkGateways"
+        ],
+        "operationId": "VirtualNetworkGateways_Resetvpnclientsharedkey",
```
Comment (sergey-shandar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json]: Could we have a better name? For example `VirtualNetworkGateways_ResetVpnClientSharedKey`

Comment (henry416) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json]: Done as requested

Comment (henry416) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json]: @sergey-shandar Done as requested

**Comment by sergey-shandar** [comment]
@henry416 we still need ARM approval for this.

**Comment by henry416** [comment]
@ravbhatnagar and @MikhailTryakhov  Can you please review this PR?

**Comment by ravbhatnagar** [comment]
looks good. signing off from ARM

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json (hunk 1 above)
Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json]: @henry416 in this case you might need to port the changes to 2018-07-01 as well. Also 2018-08-01 (working branch for that version is `Network-September-Release`)

**Comment by number213** [comment]
@henry416 Mikhail is on vacation, please contact `NRP Swagger Admins <nrp-swagger-admins@microsoft.com>`

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json (hunk 1 above)
Comment (henry416) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json]: @number213 If that is the case, I will open another PR for the 2018-08-01 version instead

**Comment by henry416** [comment]
@sergey-shandar @ravbhatnagar @number213 , as suggested, retargeted this PR for outgoing 2018 version, could you please sign off on that one? https://github.com/Azure/azure-rest-api-specs/pull/3800

**Pull request merged**